src/commonMain/kotlin/org/kosat/Clause.kt

```diff
@@ -2,7 +2,7 @@ package org.kosat
 
 typealias Lit = Int
 
-class Clause(val lits: MutableList<Lit>): Collection<Lit> {
+class Clause(val lits: MutableList<Lit>): Collection<Lit> by lits {
```

Comment:
Now, when you've switched to by-delegation, you do not need to override all the methods of `Collection`, such as `get`/`set`/`iterator`/etc :)
Please, read the documentation of by-delegation (what is that, and why it might be useful/needed), not just blindly apply it!